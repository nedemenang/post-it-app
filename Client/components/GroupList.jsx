import React, {Component} from 'react';
import {
  BrowserRouter as Router,
  Route,
  Link
} from 'react-router-dom';
import '../public/style.css';
import $ from '../public/jquery.js';
import AppActions from '../actions/AppActions';
import Group from './Group';
import {List, Card, Subheader} from 'material-ui';
//import Subheader from 'material-ui/Subheader';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import {green100, green500, green700} from 'material-ui/styles/colors';

const muiTheme = getMuiTheme({
  palette: {
    primary1Color: green500,
    primary2Color: green700,
    primary3Color: green100,
  },
}, {
  avatar: {
    borderColor: null,
  },
  //userAgent: req.headers['user-agent'],
});


class GroupList extends Component {

  handleToggle(){
    $('.group-form').slideToggle();
    //this.props.errors = '';
  };

 // groupClicked() {
   // console.log('click me jooorr')
 // }

  constructor(props){
    super(props);
    this.handleToggle = this.handleToggle.bind(this);
   // this.groupClicked = this.groupClicked.bind(this);
    this.state= {};
  }
  render(){
    //console.log(this.props.groups)
    return(
      <div className="bottomMargin">
      <MuiThemeProvider muiTheme={muiTheme}>
       <Card>
        <List>
        <Subheader><strong>Group List</strong></Subheader>
          {
            this.props.groups.map((group, i) => {
                  return <Group group={group} key={i} />
              })
          }
          </List>
          <button className="button" onClick={this.handleToggle}>Create New Group</button>
       </Card>
          </MuiThemeProvider>
      </div>

    );
  }
}

export default GroupList;