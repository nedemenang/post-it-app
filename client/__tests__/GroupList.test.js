import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import GroupList from '../components/GroupList.jsx';

describe('the group list component', () => {
  let mountedComponent;

  const str = location.href;
  const str2 = str.replace('#', ' ');

  const loggedInUser = {
    displayName: 'nnamisco',
    email: 'nedemenang@gmail.com',
    id: 'TUsUaucS5nb5kuOP6SpvTUcX9vD3',
    isAuthenticated: true,
    phoneNo: null,
    profilePic: `${str2}static/files/blank-profile-pic.png`
  };

  const group = { groupId: 'TUsUaucS5nb5kuOP6SpvTUcX9v3',
    groupName: 'Test Group',
    newMessage: false };

  const groups = [{ groupId: 'TUsUaucS5nb5kuOP6SpvTUcX9v3',
    groupName: 'Test Group',
    newMessage: false },
  { groupId: 'TUsUaucS5nb5kuOP6SpvTUcX5v3',
    groupName: 'Test Group2',
    newMessage: false },
  { groupId: 'TUsUaucS5nb5kuOP6SpvTUdX9v3',
    groupName: 'Test Group3',
    newMessage: true }];


  const groupListItem = () => {
    if (!mountedComponent) {
      mountedComponent = shallow(<GroupList groups = {groups}
      selectedGroup={group} loggedInUser = {loggedInUser}/>);
    }
    return mountedComponent;
  };

  beforeEach(() => {
    mountedComponent = undefined;
  });

  it('should render', () => {
    const component = groupListItem();
    expect(component).toBeDefined();
  });

  it('should match snapshot test', () => {
    const component = groupListItem();
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });

  it('should call function on click', () => {
    const component = shallow(<GroupList groups = {groups}
      selectedGroup={group} loggedInUser = {loggedInUser}/>);
    component.find('.button').simulate('Click');
    expect(component).toMatchSnapshot();
  });

  it('should recieve props', () => {
    expect(Object.keys(groupListItem().props()).length).toBeGreaterThan(0);
    expect(groupListItem().instance().props.groups).toEqual(groups);
    expect(groupListItem().instance().props.selectedGroup).toEqual(group);
    expect(groupListItem().instance().props.loggedInUser).toEqual(loggedInUser);
  });
});
