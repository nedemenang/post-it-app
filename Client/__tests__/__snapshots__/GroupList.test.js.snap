// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the group list component should call function on click 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="bottomMargin"
>
    <div>
        <List>
            <Subheader
                inset={false}
            >
                <strong>
                    Group List
                </strong>
            </Subheader>
            <Group
                group={
                    Object {
                        "groupId": "TUsUaucS5nb5kuOP6SpvTUcX9v3",
                        "groupName": "Test Group",
                        "newMessage": false,
                      }
                }
                loggedInUser={
                    Object {
                        "displayName": "nnamisco",
                        "email": "nedemenang@gmail.com",
                        "id": "TUsUaucS5nb5kuOP6SpvTUcX9vD3",
                        "isAuthenticated": true,
                        "phoneNo": null,
                        "profilePic": "about:blankstatic/files/blank-profile-pic.png",
                      }
                }
            />
            <Group
                group={
                    Object {
                        "groupId": "TUsUaucS5nb5kuOP6SpvTUcX5v3",
                        "groupName": "Test Group2",
                        "newMessage": false,
                      }
                }
                loggedInUser={
                    Object {
                        "displayName": "nnamisco",
                        "email": "nedemenang@gmail.com",
                        "id": "TUsUaucS5nb5kuOP6SpvTUcX9vD3",
                        "isAuthenticated": true,
                        "phoneNo": null,
                        "profilePic": "about:blankstatic/files/blank-profile-pic.png",
                      }
                }
            />
            <Group
                group={
                    Object {
                        "groupId": "TUsUaucS5nb5kuOP6SpvTUdX9v3",
                        "groupName": "Test Group3",
                        "newMessage": true,
                      }
                }
                loggedInUser={
                    Object {
                        "displayName": "nnamisco",
                        "email": "nedemenang@gmail.com",
                        "id": "TUsUaucS5nb5kuOP6SpvTUcX9vD3",
                        "isAuthenticated": true,
                        "phoneNo": null,
                        "profilePic": "about:blankstatic/files/blank-profile-pic.png",
                      }
                }
            />
        </List>
        <button
            className="button"
            onClick={[Function]}
        >
            Create New Group
        </button>
    </div>
</div>,
  "nodes": Array [
    <div
      className="bottomMargin"
>
      <div>
            <List>
                  <Subheader
                        inset={false}
                  >
                        <strong>
                              Group List
                        </strong>
                  </Subheader>
                  <Group
                        group={
                              Object {
                                    "groupId": "TUsUaucS5nb5kuOP6SpvTUcX9v3",
                                    "groupName": "Test Group",
                                    "newMessage": false,
                                  }
                        }
                        loggedInUser={
                              Object {
                                    "displayName": "nnamisco",
                                    "email": "nedemenang@gmail.com",
                                    "id": "TUsUaucS5nb5kuOP6SpvTUcX9vD3",
                                    "isAuthenticated": true,
                                    "phoneNo": null,
                                    "profilePic": "about:blankstatic/files/blank-profile-pic.png",
                                  }
                        }
                  />
                  <Group
                        group={
                              Object {
                                    "groupId": "TUsUaucS5nb5kuOP6SpvTUcX5v3",
                                    "groupName": "Test Group2",
                                    "newMessage": false,
                                  }
                        }
                        loggedInUser={
                              Object {
                                    "displayName": "nnamisco",
                                    "email": "nedemenang@gmail.com",
                                    "id": "TUsUaucS5nb5kuOP6SpvTUcX9vD3",
                                    "isAuthenticated": true,
                                    "phoneNo": null,
                                    "profilePic": "about:blankstatic/files/blank-profile-pic.png",
                                  }
                        }
                  />
                  <Group
                        group={
                              Object {
                                    "groupId": "TUsUaucS5nb5kuOP6SpvTUdX9v3",
                                    "groupName": "Test Group3",
                                    "newMessage": true,
                                  }
                        }
                        loggedInUser={
                              Object {
                                    "displayName": "nnamisco",
                                    "email": "nedemenang@gmail.com",
                                    "id": "TUsUaucS5nb5kuOP6SpvTUcX9vD3",
                                    "isAuthenticated": true,
                                    "phoneNo": null,
                                    "profilePic": "about:blankstatic/files/blank-profile-pic.png",
                                  }
                        }
                  />
            </List>
            <button
                  className="button"
                  onClick={[Function]}
            >
                  Create New Group
            </button>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <GroupList
        groups={
                Array [
                        Object {
                          "groupId": "TUsUaucS5nb5kuOP6SpvTUcX9v3",
                          "groupName": "Test Group",
                          "newMessage": false,
                        },
                        Object {
                          "groupId": "TUsUaucS5nb5kuOP6SpvTUcX5v3",
                          "groupName": "Test Group2",
                          "newMessage": false,
                        },
                        Object {
                          "groupId": "TUsUaucS5nb5kuOP6SpvTUdX9v3",
                          "groupName": "Test Group3",
                          "newMessage": true,
                        },
                      ]
        }
        loggedInUser={
                Object {
                        "displayName": "nnamisco",
                        "email": "nedemenang@gmail.com",
                        "id": "TUsUaucS5nb5kuOP6SpvTUcX9vD3",
                        "isAuthenticated": true,
                        "phoneNo": null,
                        "profilePic": "about:blankstatic/files/blank-profile-pic.png",
                      }
        }
        selectedGroup={
                Object {
                        "groupId": "TUsUaucS5nb5kuOP6SpvTUcX9v3",
                        "groupName": "Test Group",
                        "newMessage": false,
                      }
        }
/>,
      "_debugID": 5,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": GroupList {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "handleToggle": [Function],
        "mounted": false,
        "props": Object {
          "groups": Array [
            Object {
              "groupId": "TUsUaucS5nb5kuOP6SpvTUcX9v3",
              "groupName": "Test Group",
              "newMessage": false,
            },
            Object {
              "groupId": "TUsUaucS5nb5kuOP6SpvTUcX5v3",
              "groupName": "Test Group2",
              "newMessage": false,
            },
            Object {
              "groupId": "TUsUaucS5nb5kuOP6SpvTUdX9v3",
              "groupName": "Test Group3",
              "newMessage": true,
            },
          ],
          "loggedInUser": Object {
            "displayName": "nnamisco",
            "email": "nedemenang@gmail.com",
            "id": "TUsUaucS5nb5kuOP6SpvTUcX9vD3",
            "isAuthenticated": true,
            "phoneNo": null,
            "profilePic": "about:blankstatic/files/blank-profile-pic.png",
          },
          "selectedGroup": Object {
            "groupId": "TUsUaucS5nb5kuOP6SpvTUcX9v3",
            "groupName": "Test Group",
            "newMessage": false,
          },
        },
        "refs": Object {},
        "state": Object {},
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 3,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="bottomMargin"
>
          <div>
                    <List>
                              <Subheader
                                        inset={false}
                              >
                                        <strong>
                                                  Group List
                                        </strong>
                              </Subheader>
                              <Group
                                        group={
                                                  Object {
                                                            "groupId": "TUsUaucS5nb5kuOP6SpvTUcX9v3",
                                                            "groupName": "Test Group",
                                                            "newMessage": false,
                                                          }
                                        }
                                        loggedInUser={
                                                  Object {
                                                            "displayName": "nnamisco",
                                                            "email": "nedemenang@gmail.com",
                                                            "id": "TUsUaucS5nb5kuOP6SpvTUcX9vD3",
                                                            "isAuthenticated": true,
                                                            "phoneNo": null,
                                                            "profilePic": "about:blankstatic/files/blank-profile-pic.png",
                                                          }
                                        }
                              />
                              <Group
                                        group={
                                                  Object {
                                                            "groupId": "TUsUaucS5nb5kuOP6SpvTUcX5v3",
                                                            "groupName": "Test Group2",
                                                            "newMessage": false,
                                                          }
                                        }
                                        loggedInUser={
                                                  Object {
                                                            "displayName": "nnamisco",
                                                            "email": "nedemenang@gmail.com",
                                                            "id": "TUsUaucS5nb5kuOP6SpvTUcX9vD3",
                                                            "isAuthenticated": true,
                                                            "phoneNo": null,
                                                            "profilePic": "about:blankstatic/files/blank-profile-pic.png",
                                                          }
                                        }
                              />
                              <Group
                                        group={
                                                  Object {
                                                            "groupId": "TUsUaucS5nb5kuOP6SpvTUdX9v3",
                                                            "groupName": "Test Group3",
                                                            "newMessage": true,
                                                          }
                                        }
                                        loggedInUser={
                                                  Object {
                                                            "displayName": "nnamisco",
                                                            "email": "nedemenang@gmail.com",
                                                            "id": "TUsUaucS5nb5kuOP6SpvTUcX9vD3",
                                                            "isAuthenticated": true,
                                                            "phoneNo": null,
                                                            "profilePic": "about:blankstatic/files/blank-profile-pic.png",
                                                          }
                                        }
                              />
                    </List>
                    <button
                              className="button"
                              onClick={[Function]}
                    >
                              Create New Group
                    </button>
          </div>
</div>,
        "_debugID": 6,
        "_renderedOutput": <div
          className="bottomMargin"
>
          <div>
                    <List>
                              <Subheader
                                        inset={false}
                              >
                                        <strong>
                                                  Group List
                                        </strong>
                              </Subheader>
                              <Group
                                        group={
                                                  Object {
                                                            "groupId": "TUsUaucS5nb5kuOP6SpvTUcX9v3",
                                                            "groupName": "Test Group",
                                                            "newMessage": false,
                                                          }
                                        }
                                        loggedInUser={
                                                  Object {
                                                            "displayName": "nnamisco",
                                                            "email": "nedemenang@gmail.com",
                                                            "id": "TUsUaucS5nb5kuOP6SpvTUcX9vD3",
                                                            "isAuthenticated": true,
                                                            "phoneNo": null,
                                                            "profilePic": "about:blankstatic/files/blank-profile-pic.png",
                                                          }
                                        }
                              />
                              <Group
                                        group={
                                                  Object {
                                                            "groupId": "TUsUaucS5nb5kuOP6SpvTUcX5v3",
                                                            "groupName": "Test Group2",
                                                            "newMessage": false,
                                                          }
                                        }
                                        loggedInUser={
                                                  Object {
                                                            "displayName": "nnamisco",
                                                            "email": "nedemenang@gmail.com",
                                                            "id": "TUsUaucS5nb5kuOP6SpvTUcX9vD3",
                                                            "isAuthenticated": true,
                                                            "phoneNo": null,
                                                            "profilePic": "about:blankstatic/files/blank-profile-pic.png",
                                                          }
                                        }
                              />
                              <Group
                                        group={
                                                  Object {
                                                            "groupId": "TUsUaucS5nb5kuOP6SpvTUdX9v3",
                                                            "groupName": "Test Group3",
                                                            "newMessage": true,
                                                          }
                                        }
                                        loggedInUser={
                                                  Object {
                                                            "displayName": "nnamisco",
                                                            "email": "nedemenang@gmail.com",
                                                            "id": "TUsUaucS5nb5kuOP6SpvTUcX9vD3",
                                                            "isAuthenticated": true,
                                                            "phoneNo": null,
                                                            "profilePic": "about:blankstatic/files/blank-profile-pic.png",
                                                          }
                                        }
                              />
                    </List>
                    <button
                              className="button"
                              onClick={[Function]}
                    >
                              Create New Group
                    </button>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <GroupList
    groups={
        Array [
            Object {
              "groupId": "TUsUaucS5nb5kuOP6SpvTUcX9v3",
              "groupName": "Test Group",
              "newMessage": false,
            },
            Object {
              "groupId": "TUsUaucS5nb5kuOP6SpvTUcX5v3",
              "groupName": "Test Group2",
              "newMessage": false,
            },
            Object {
              "groupId": "TUsUaucS5nb5kuOP6SpvTUdX9v3",
              "groupName": "Test Group3",
              "newMessage": true,
            },
          ]
    }
    loggedInUser={
        Object {
            "displayName": "nnamisco",
            "email": "nedemenang@gmail.com",
            "id": "TUsUaucS5nb5kuOP6SpvTUcX9vD3",
            "isAuthenticated": true,
            "phoneNo": null,
            "profilePic": "about:blankstatic/files/blank-profile-pic.png",
          }
    }
    selectedGroup={
        Object {
            "groupId": "TUsUaucS5nb5kuOP6SpvTUcX9v3",
            "groupName": "Test Group",
            "newMessage": false,
          }
    }
/>,
}
`;

exports[`the group list component should match snapshot test 1`] = `
<div
  className="bottomMargin"
>
  <div>
    <List>
      <Subheader
        inset={false}
      >
        <strong>
          Group List
        </strong>
      </Subheader>
      <Group
        group={
          Object {
            "groupId": "TUsUaucS5nb5kuOP6SpvTUcX9v3",
            "groupName": "Test Group",
            "newMessage": false,
          }
        }
        loggedInUser={
          Object {
            "displayName": "nnamisco",
            "email": "nedemenang@gmail.com",
            "id": "TUsUaucS5nb5kuOP6SpvTUcX9vD3",
            "isAuthenticated": true,
            "phoneNo": null,
            "profilePic": "about:blankstatic/files/blank-profile-pic.png",
          }
        }
      />
      <Group
        group={
          Object {
            "groupId": "TUsUaucS5nb5kuOP6SpvTUcX5v3",
            "groupName": "Test Group2",
            "newMessage": false,
          }
        }
        loggedInUser={
          Object {
            "displayName": "nnamisco",
            "email": "nedemenang@gmail.com",
            "id": "TUsUaucS5nb5kuOP6SpvTUcX9vD3",
            "isAuthenticated": true,
            "phoneNo": null,
            "profilePic": "about:blankstatic/files/blank-profile-pic.png",
          }
        }
      />
      <Group
        group={
          Object {
            "groupId": "TUsUaucS5nb5kuOP6SpvTUdX9v3",
            "groupName": "Test Group3",
            "newMessage": true,
          }
        }
        loggedInUser={
          Object {
            "displayName": "nnamisco",
            "email": "nedemenang@gmail.com",
            "id": "TUsUaucS5nb5kuOP6SpvTUcX9vD3",
            "isAuthenticated": true,
            "phoneNo": null,
            "profilePic": "about:blankstatic/files/blank-profile-pic.png",
          }
        }
      />
    </List>
    <button
      className="button"
      onClick={[Function]}
    >
      Create New Group
    </button>
  </div>
</div>
`;
